subsys/net/ip/ipv6.c

```diff
@@ -588,7 +588,8 @@ enum net_verdict net_ipv6_input(struct net_pkt *pkt, bool is_loopback)
 
 	if (net_ipv6_is_addr_mcast((struct in6_addr *)hdr->dst) &&
 	    !(net_ipv6_is_addr_mcast_iface((struct in6_addr *)hdr->dst) ||
-	      net_ipv6_is_addr_mcast_link_all_nodes((struct in6_addr *)hdr->dst))) {
+	      net_ipv6_is_addr_mcast_link_all_nodes((struct in6_addr *)hdr->dst)) &&
+	    !(net_ipv6_is_addr_solicited_node((struct in6_addr *)hdr->dst))) {
```

Comment:
Copying my comment from #71471

I don't think I understand why would Solicited Node Multicast addresses need special treatment here, especially that with this change we'd pass multicast packet addressing any Solicited Node Multicast address, not necessarily our own. The Solicited Node Multicast address (our own) should be registered on the interface, so the checks below should pass.

Perhaps you're facing some configuration issue, I. e. too little mcast address buffers in the system (CONFIG_NET_IF_MCAST_IPV6_ADDR_COUNT), so the solicited-node multicast does not end up registered on the interface?

Reply:
Hi Rlubos,

You mean I should call net_if_ipv6_addr_add to join Solicited Node Multicast address after interface up, right?

Reply:
Solicited Node Multicast address should be added and joined automatically when you add an unicast address with `net_if_ipv6_addr_add()`. However, as mentioned earlier, if for instance you have too little mcast address buffers (configured with `CONFIG_NET_IF_MCAST_IPV6_ADDR_COUNT`) it may fail to be added.

If you have access to net shell, it'd be easiest to check with `net iface` command to see what multicast addresses are configured on the interface. If the solicited node multicast is missing, it's likely an configuration issue.